app/models/user.rb

```diff
@@ -11,15 +11,14 @@ class User < ApplicationRecord
   has_many :taxes, through: :activities
   has_many :organizations, -> { distinct }, through: :activities
   validates :email, presence: true, uniqueness: true
-  validates :password, presence: true, confirmation: true, length: {minimum: 6}
-  validates :password_confirmation, presence: true
+  validates :password, presence: true, confirmation: true, length: {minimum: 6}, unless: :provider?
+  validates :password_confirmation, presence: true, unless: :provider?
 
   def self.from_omniauth(auth)
     where(provider: auth.provider, uid: auth.uid).first_or_create do |user|
       user.provider = auth.provider
       user.uid = auth.uid
-      user.email = auth.info.email
-      user.password = Devise.friendly_token[0, 20]
+      user.email = auth.info.email || auth.info.nickname + '_' + auth.provider
```

Comment:
why do you want to assign this to `email`? the email you want to build does not exist.

if you want to store nickname, you can have special fields for that.